File: sandbox/api/models/Transaction.js
const mongoose = require("mongoose");
const Schema = mongoose.Schema;

const Transacton = new mongoose.Schema(
    {
        transactionId: {
            type: String,
            unique: true,
            required: [true, 'Please specify the ID']
        },
        transactionDescription: {
            type: String,
            required: [true, 'Please specify the description']
        },
        transactionType: {
            type: String,
            required: [true, 'Please specify the type']
        },
        transactionDate: {
            type: Date,
            default: Date.now
        },
        transactionAmount: {
            type: Number,
            required: [true, 'Please specify the amount']
        }
    }
);

module.exports = mongoose.model('Transacton', Transacton)

File: ios/app/app/API/Models/Transaction.swift
//
//  Transaction.swift
//  app
//
//  Created by wenzhong zheng on 2021-01-10.
//

import Foundation

struct Transaction: Codable {
    let transactionId: String
    let transactionDescription: String
    let transactionType: String
    let transactionDate: Date
    let transactionAmount: Double
}
